### DISCUSSION TITLE
Subprojects repeatedly executed when compiling

### DISCUSSION THREAD

**Discussion by vpoinot** [mesonbuild/meson#13939] [Q&A]
I have 4 subprojects in my meson project.
Every time I run `meson compile`, meson goes into each subproject and outputs:
```
Executing subproject subproject-1
...
Executing subproject subproject-2
...
etc.
```
And for each of them, it looks like it executes the meson.build file found there. Is this behaviour expected? 

Of course, it makes sense, in a way, but my intent was that these third-party dependencies would be unwrapped (they come from wrapdb), compiled once, and then be done with them: I mean, these dependencies don't change, so in my case at least, there is no need to go over them every time, only my own project is worth checking for changes and recompiling.

Is there any way to achieve this (i.e. just recompile my project, don't check subprojects)?

**Comment by vpoinot** [comment]
Hmm, actually this behaviour is not systematic. Forget it for now.

**Comment by eli-schwartz** [comment]
This status message should only be coming from `meson setup`, including when the entire build configuration has to be reconfigured. The implication would be that ninja believes your `meson.build` file(s) have been modified, either the main project or a subproject, and has reconfigured *everything* -- i.e. this isn't about subprojects specifically.

Maybe a full log of your console session would help me understand your issue better.

**Reply by vpoinot** [reply]
Indeed, I saw this behaviour because I was experimenting with my meson.build, and every change in this file would imply the behaviour I described. If I don't modify meson.build, then nothing happens.

Sorry, still new to meson, need to get my head around it :-)

**Reply by eli-schwartz** [reply]
Ok sure, no problem. :)

We don't cache entire subprojects for reconfiguring, as the results of one subproject can cause logic changes inside another subproject and require re-evaluating if/else branches. Although we do cache most of the slowest parts of a reconfigure, in particular, `dependency()` calls to system packages as well as `cc.has_function()` and similar. All of these require running slow subprocesses. A subproject can run entirely in-process and may be quite fast, so it's not a big deal to re-evaluate it for correctness.
